tests/functional/aws-node-sdk/test/multipleBackend/delete.js

```diff
@@ -0,0 +1,64 @@
+import assert from 'assert';
+import withV4 from '../support/withV4';
+import BucketUtility from '../../lib/utility/bucket-util';
+
+const bucket = 'buckettestmultiplebackenddelete';
+const memObject = 'memObject';
+const fileObject = 'fileObject';
+
+describe('Multiple backend delete', () => {
+    withV4(sigCfg => {
+        let bucketUtil;
+        let s3;
+
+        before(() => {
+            process.stdout.write('Creating bucket\n');
+            bucketUtil = new BucketUtility('default', sigCfg);
+            s3 = bucketUtil.s3;
+            return s3.createBucketAsync({ Bucket: bucket })
+            .catch(err => {
+                process.stdout.write(`Error creating bucket: ${err}\n`);
+                throw err;
+            })
+            .then(() => {
+                process.stdout.write('Putting object to mem\n');
+                const params = { Bucket: bucket, Key: memObject,
```

Comment:
If you're not sending a body, you're really using the 0 byte object path and there is nothing to delete.  we should test both paths (with and without a data body)